concept/type_/type_manager/type_reader.rs

```diff
@@ -55,6 +58,7 @@ use crate::{
 // non-instantiable
 pub enum TypeReader {}
 
+// TODO: this file needs to use StorageCounters correctly
```

Comment:
I've left this TODO hanging as pulling the storage counters here would be a _massive_ change, and most of these are very cheap, so not worth the time investment.

Reply:
Agree as well!